### PR TITLE
More functional

### PR DESCRIPTION
**Pull request by zdrve**
Less passing of data via instance variables.

The name `internal_plan` is a bit clunky, but the obvious name `plan` is taken, repeatedly:

 * the "internal" plan (formerly `@create, @update, @delete`)
 * the "external" plan, as returned by ...
 * ... the method `#plan`

Just for bonus giggles, `plan` is of course both a noun and a verb.

We could simplify things a lot by making the external and internal plans the same, and then the name "plan" would work for everything.

### CODE REVIEW

File: lib/kennel/syncer.rb [hunk 1]
```diff
@@ -164,14 +167,14 @@ def initialize
         @attribute_differ = AttributeDiffer.new
       end
 
-      def display(create, update, delete)
+      def display(internal_plan)
         Kennel.out.puts "Plan:"
-        if create.empty? && update.empty? && delete.empty?
+        if internal_plan.values.all?(&:empty?)
```
Comment (grosser) [inline lib/kennel/syncer.rb:172]:
```suggestion
        if noop?
```

Comment (zdrve) [reply lib/kennel/syncer.rb:172]: It's in a different class now 😅

Comment (grosser) [reply lib/kennel/syncer.rb:172]: then it should have the `noop?` method ?

**Review by grosser — APPROVED** [review]
the state reduction is nice 👍

**Comment by zdrve** [comment]
Yeah the core is getting a lot easier to understand now!

**Pull request merged**